internal/controller/databaseclusterrestore_controller.go

```diff
@@ -730,3 +721,164 @@ func (r *DatabaseClusterRestoreReconciler) ReconcileWatchers(ctx context.Context
+		Name:      obj.GetName(),
+	}
+
+	if err := r.Get(ctx, namespacedName, pgRestore); err != nil {
+		// if such upstream restore is not found - do nothing
+		if k8serrors.IsNotFound(err) {
+			return nil
+		}
+
+		return err
+	}
+
+	pg := &pgv2.PerconaPGCluster{}
+	if err := r.Get(ctx, types.NamespacedName{Namespace: obj.GetNamespace(), Name: pgRestore.Spec.PGCluster}, pg); err != nil {
+		// if such upstream cluster is not found - do nothing
+		if k8serrors.IsNotFound(err) {
+			return nil
+		}
+
+		return err
+	}
+
+	storages := &everestv1alpha1.BackupStorageList{}
+	if err := r.List(ctx, storages, &client.ListOptions{Namespace: namespacedName.Namespace}); err != nil {
+		// if no backup storages found - do nothing
+		if k8serrors.IsNotFound(err) {
+			return nil
+		}
+
+		return err
+	}
+
+	restore := &everestv1alpha1.DatabaseClusterRestore{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      obj.GetName(),
+			Namespace: obj.GetNamespace(),
+		},
+	}
+
+	err := r.Get(ctx, namespacedName, restore)
+	// if such everest restore already exists - do nothing
+	if err == nil {
+		return nil
+	}
+
+	if !k8serrors.IsNotFound(err) {
+		return err
+	}
+
+	restore.Spec.DBClusterName = pgRestore.Spec.PGCluster
+
+	cluster := &everestv1alpha1.DatabaseCluster{}
+
+	err = r.Get(ctx, types.NamespacedName{Name: pgRestore.Spec.PGCluster, Namespace: pgRestore.Namespace}, cluster)
+	if err != nil {
+		return err
+	}
+	// The restore that we try to create when pg-restore CR appears is always a bootstrap restore
+	// so we take the DataSource details from the dbCluster.DataSource.
+	if cluster.Spec.DataSource != nil { //nolint:nestif
+		// if the bootstrap restore is triggered by DataImport - delete the upstream pg-restore (it's excessive)
+		// and do not create a dbr for it accordingly
+		if cluster.Spec.DataSource.DataImport != nil {
+			err = r.Delete(ctx, obj)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		}
+
+		if cluster.Spec.DataSource.DBClusterBackupName != "" {
+			restore.Spec.DataSource.DBClusterBackupName = cluster.Spec.DataSource.DBClusterBackupName
+		}
+
+		if cluster.Spec.DataSource.BackupSource != nil {
+			restore.Spec.DataSource.BackupSource = &everestv1alpha1.BackupSource{
+				Path:              "",
+				BackupStorageName: cluster.Spec.DataSource.BackupSource.BackupStorageName,
+			}
+		}
+
+		if cluster.Spec.DataSource.PITR != nil {
+			restore.Spec.DataSource.PITR = &everestv1alpha1.PITR{
+				Type: cluster.Spec.DataSource.PITR.Type,
+				Date: cluster.Spec.DataSource.PITR.Date,
+			}
+		}
+	}
+
+	restore.Labels = map[string]string{
+		consts.DatabaseClusterNameLabel: pgRestore.Spec.PGCluster,
+	}
+	if err = controllerutil.SetControllerReference(cluster, restore, r.Scheme); err != nil {
+		return err
+	}
+
+	return r.Create(ctx, restore)
+}
+
+func (r *DatabaseClusterRestoreReconciler) tryDeleteDBRestore(ctx context.Context, obj client.Object) {
+	logger := log.FromContext(ctx)
+	logger.Info("DELETE!!!!")
```

Comment:
missed leftover

Reply:
pushed to the loadbalancerconfig feature branch https://github.com/percona/everest-operator/pull/864/commits/306005cf1eee2a8cf34bc63467361c8c1332c482